src/main/docs/guide/httpServer/customArgumentBinding.adoc

```diff
@@ -0,0 +1,59 @@
+
+== Custom argument binding
+
+Micronaut uses an api:http.bind.binders.ArgumentBinderRegistry[ArgumentBinderRegistry] to look up api:core.bind.ArgumentBinder[ArgumentBinder]s
+capable of binding to the arguments in the controller methods. The default implementation will look for an annotation on
+the argument that is, itself, annotated with `@Bindable`. If one exists the argument binder registry will search for an
+argument binder that supports that annotation.
+
+If no fitting annotation is found Micronaut will try to find an argument binder that supports the argument's type.
+
+An api:core.bind.ArgumentBinder[ArgumentBinder] can be registered as a bean. In this case it is only present when a
+api:context.BeanContext[BeanContext] is initialized.
+
+Binders are called by the same server logic that calls the api:io.micronaut.http.annotation.Controller[Controller]. Once
+the request is received and the argument is resolved in the api:http.bind.binders.ArgumentBinderRegistry[ArgumentBinderRegistry]
+api:io.micronaut.http.filter.HttpFilter[HttpFilter]s are called before returning results to the Controller. Your
+api:io.micronaut.core.bind.BindingResult[BindingResult] may depend on those being initialized before returning the binding
+result.
+
+A binding result of the class you are binding will be returned from the bind method in the implementation written. The
+result will be classified as either `SATISFIED` or `UNSATISFIED`. Returning a binding of the type `EMPTY` will be considered
+SATISFIED. If the returned binding is UNSATISFIED the registry will continue to attempt to find argument binders that match
+the argument.
+
+If the binding result throws an exception that is not handled it will bubble up to the api:io.micronaut.http.annotation.Controller[Controller]
+where it can be caught by an api:io.micronaut.http.annotation.Error[Error] handler.
```

Comment:
In general this should be discouraged. Instead errors should be added to the conversion context through context.reject. That happens automatically for conversion errors

Reply:
I'm not 100% sure I understand this one but I have added some exception handling to the snippets since that sounds like something we'd want to demonstrate.